pkg/kubectl/cmd/patch.go

```diff
@@ -180,6 +180,7 @@ func RunPatch(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 		if !options.Local {
 			dataChangedMsg := "not patched"
 			helper := resource.NewHelper(client, mapping)
+			emptyPatchObj, err := helper.Patch(namespace, name, patchType, []byte("{}"))
```

Comment:
Looking at the bug description this doesn't look like what @liggitt suggested as a solution.